feature_tests/cpp/include/OptionInputStruct.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef OptionInputStruct_HPP
+#define OptionInputStruct_HPP
+
+#include "OptionInputStruct.d.hpp"
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <memory>
+#include <optional>
+#include "OptionEnum.hpp"
+#include "diplomat_runtime.hpp"
+
+
+namespace diplomat {
+namespace capi {
+    extern "C" {
+    
+    
+    } // extern "C"
+} // namespace capi
+} // namespace
+
+
+inline diplomat::capi::OptionInputStruct OptionInputStruct::AsFFI() const {
+  return diplomat::capi::OptionInputStruct {
+    /* .a = */ a.has_value() ? (diplomat::capi::OptionU8{ { a.value() }, true }) : (diplomat::capi::OptionU8{ {}, false }),
```

Comment:
Question/Suggestion: why is this code inlined instead of invoking `OptionU8::AsFFI` ?

Reply:
I'm confused, there is no such method anywhere.

Reply:
My suggestion was that such a method should be created and called here.

Reply:
This is a type from the C header: adding method to it would involve more C-CPP splits, or more duplicated code listing all of the primitives.  The code here seems rather simple, too, I don't think there's much value in trying to do it as a function.